### PR TITLE
`Assessment`: Improved assesment view for instructors

### PR DESCRIPTION
**Pull request by ole-ve**
#### General

- [ ] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [X] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.ase.in.tum.de/dev/guidelines/language-guidelines/).
- [X] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
This PR is linked to #4176.
After assessing a single submission, the button for importing this submission as an example submission appears directly under the cursor for submitting the assessment and cause instructors to accidently click on it.

### Description
The button position for importing as example submission has been moved. In addition a dialog window has been added that opens after clicking the button in order to confirm the importing process.

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Student
- Text Exercise with all dates set, so that one has enough time to start and submit the exercise and asses

1. Log in to Artemis
2. Navigate to course overview
3. Start the created text exercises
4. Submit any text
5. Wait for the submission period to be over
6. Navigate to course management, and the corresponding text exercise
7. Open submission dashboard
8. Click the button "Assess submission"
9. Add Feedback and submit
10. The buttons should be aligned correctly. When clicking the import button, the instructor should be prompted with a verification dialog. If aborted, the submission should not be imported. If approved, the submission should be imported.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
assessment-header.component.ts: 92.3% | 78.26%

### Screenshots
Before:
Bildschirmfoto 2021-11-30 um 00 05 19

After:
Bildschirmfoto 2021-11-30 um 00 04 38

Note: This PR hotfixes the input value 'isAtLeastInstructor' for the component assessment-layout.component.ts. As this value is redundantly defined as an attribute in the input object exercise, the input 'isAtLeastInstructor' should be replaced with the usage of the attribute in exercise. Follow up in #4428.

### CODE REVIEW

**Review by willergeorg — APPROVED** [review]
Tested on TS1 and works as expected. Code changes look reasonable

**Review by Malyuk-A — APPROVED** [review]
Tested on TS1. Workes as described 👍
I just would like to hear your opinion about 2 things:
a) The "import" strings seem to actually be exporting something instead of importing it. So do you think it would be a good idea to change all "import" strings into "export"?
b) This might be tricky (because CSS and alignment can be very frustrating) but if you could remove the right margin for the very right button it should perfectly align with the rest.

**Review by VinnavanSembian — APPROVED** [review]
Tested on ts3. The buttons are aligned correctly.

**Review by mallgaier — APPROVED** [review]
Tested together with @cremertim @Strohgelaender @Bofan-Zhu and @lBruchmann on TS3. The functionality was implemented as described and the buttons were swapped.

**Review by cremertim — APPROVED** [review]
tested on ts3, everything works as expected. Great job 👍

**Review by Strohgelaender — APPROVED** [review]
Tested in testing session with @mallgaier @lBruchmann @cremertim @VinnavanSembian @Bofan-Zhu. Worked as expected, I only have a small code comment:

**Comment by Bofan-Zhu** [comment]
Tested on ts3 and everything worked perfectly fine.

**Review by lBruchmann — DISMISSED** [review]
Tested on ts3. I followed the testing steps. The buttons are displayed as expected.

**Review by Bofan-Zhu — DISMISSED** [review]
Tested on t3 and everything works as expected.

**Comment by ole-ve** [comment]
> a) [...] So do you think it would be a good idea to change all "import" strings into "export"?

I think 'import' may fit better than 'export'. In my understanding, 'exporting' suggests exporting some files from the Artemis server to the outside of Artemis (e.g. local computer). I agree, that 'import' may be a little bit confusing here but 'export' even more imo.

> b) This might be tricky (because CSS and alignment can be very frustrating) but if you could remove the right margin for the very right button it should perfectly align with the rest.

Changed in 39cca73.

File: src/main/webapp/app/assessment/assessment-header/assessment-header.component.ts [hunk 1]
```diff
@@ -60,7 +62,7 @@ export class AssessmentHeaderComponent {
         this.highlightDifferencesChange.emit(this.highlightDifferences);
     }
 
-    constructor(public textAssessmentAnalytics: TextAssessmentAnalytics, protected route: ActivatedRoute) {
+    constructor(public textAssessmentAnalytics: TextAssessmentAnalytics, protected route: ActivatedRoute, private translateService: TranslateService) {
```
[the remark this answers is not in the document]
Comment (ole-ve) [reply src/main/webapp/app/assessment/assessment-header/assessment-header.component.ts]: Changed in 0339681.

**Comment by ole-ve** [comment]
I changed all client-side translations from 'import' to 'use as' example submission in f23addd.
I did not change the server-side code, where 'import' is still used but not visible to the end-user. Should I also incorporate changes on the server side (functions, rest endpoints, etc.)?

**Review by Strohgelaender — DISMISSED** [review]
re-approve

**Review by baranimo7 — DISMISSED** [review]
Tested on ts1. The order of the buttoms is correct and using the submission as an example submission works as described.

File: src/main/webapp/app/assessment/assessment-header/assessment-header.component.html [hunk 2]
```diff
@@ -172,6 +172,15 @@ <h3 class="top-container flex-wrap flex-lg-nowrap">
                 <fa-icon *ngIf="submitBusy" [icon]="faSpinner" [spin]="true">&nbsp;</fa-icon>
                 <span jhiTranslate="artemisApp.assessment.button.overrideAssessment">Override Assessment</span>
             </button>
+            <button
+                class="btn m-1 btn-primary"
+                id="useAsExampleSubmission"
+                *ngIf="result?.completionDate && isAtLeastInstructor && (exercise!.type === ExerciseType.MODELING || exercise!.type === ExerciseType.TEXT)"
+                (click)="onUseAsExampleSolutionClicked()"
+            >
+                <fa-icon *ngIf="submitBusy" [icon]="'spinner'" [spin]="true"></fa-icon>
```
Comment (Malyuk-A) [inline src/main/webapp/app/assessment/assessment-header/assessment-header.component.html:181]: Please don't forget to think about the new way we handle icons.
Have a look here if it is needed here 
https://github.com/ls1intum/Artemis/pull/4383

Comment (ole-ve) [reply src/main/webapp/app/assessment/assessment-header/assessment-header.component.html]: Agree, changed.

**Review by willergeorg — APPROVED** [review]
Tested on TS1. Works as expected. Code looks good to me as well.

**Review by valeriebucher — APPROVED** [review]
Tested on ts1, worked as described!

**Review by Hialus — APPROVED** [review]
Reviewed the code.
Looks good to me 👍🏻

**Pull request merged**